xml-conduit/src/Text/XML/Stream/Parse.hs

```diff
@@ -659,11 +660,49 @@ parseContent (ParseSettings decodeEntities _ decodeIllegalCharacters _) breakDou
     case toValidXmlChar n <|> decodeIllegalCharacters n of
       Nothing -> fail "Invalid character from decimal character reference."
       Just c  -> return $ ContentText $ T.singleton c
-  parseTextContent = ContentText <$> takeWhile1 valid <?> "text content"
+
+  -- Turns @\r\n@ and @\r@ into @\n@. See
+  -- <https://www.w3.org/TR/REC-xml/#sec-line-ends>.
+  parseTextContent = do
+    firstChunk <- takeWhile valid
+    mbC <- peekChar
+    case mbC of
+      Nothing ->
+        exit firstChunk
+      Just '\r' ->
+        parseTextContentAcc [firstChunk]
+      Just _ ->
+        exit firstChunk
+
+  parseTextContentAcc chunks = do
+    mbC <- peekChar
+    case mbC of
+      Nothing -> exit' chunks
+      Just '\r' -> do
+        _ <- anyChar
+        mbD <- peekChar
+        chunk <- takeWhile valid
+        case mbD of
+          Nothing -> exit' chunks
```

Comment:
I expected to read `Nothing -> exit' $ "\n" : chunks` instead, but then I realized this `case` instance should never happen... Do we agree that this should never happen ? (Trying to make sure I understood how this implementation works.)